content/docs/configuration-services.md

```diff
@@ -1,34 +1,33 @@
 ---
 title: "Configuration via Services"
 ---
-Langium supports the configuration of most aspects of your language and language server via a set of *Services*. Those services are organized into *Modules*, which are essentially mappings from a service name to its implementation.
 
-We can separate modules into two main categories:
-1. ***Shared modules*** which contain services that are shared across all languages.
-2. ***Language specific modules*** which contain services specific to one language. 
+Langium supports the configuration of most aspects of your language and language server via a set of *services*. Those services are configured by *modules*, which are essentially mappings from a service name to its implementation.
 
-## Shared Modules
-The *shared modules* contain services shared across all languages:
-* The `ServiceRegistry` is responsible for registering and accessing the different services.
-* The `Connection` service sends messages to the client and register message handlers for incoming messages.
-* Services to handle *Langium Documents* and *Text Documents*.
-* Implementations of the `AstReflection` service (one implementation per language) to access the structure of the AST.
+We can separate services and modules into two main categories:
 
-## Language Specific Modules
-The *language specific modules* contain all services which are specific to one language:
+#### Shared Services
+The *shared services* are services that are shared across all Langium languages. In many applications there is only one Langium language, but the overall structure of the services is the same.
+* The `ServiceRegistry` is responsible for registering and accessing the different languages and their services.
+* The `Connection` service is used in a language server context; it sends messages to the client and registers message handlers for incoming messages.
+* The `AstReflection` service provides access the structure of the AST types.
+* Shared services involved in the document lifecycle (future documentation)
+
+#### Language Specific Services
+The *language specific services* are services specific to one Langium language and isolated from other languages.
 * Services for [LSP features](#language-server-protocol)
 * Services involved in the document lifecycle (future documentation)
```

Comment:
Just after a quick look at the 'createModule functions', aren't all document services shared?

Reply:
Services such as `Linker` and `DocumentValidator` are language-specific and also involved in the document lifecycle.